src/overlays/actors/ovl_Bg_Mori_Hashigo/z_bg_mori_hashigo.c

```diff
@@ -20,39 +40,255 @@ const ActorInit Bg_Mori_Hashigo_InitVars = {
+    return 1;
+}
+
+void BgMoriHashigo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriHashigo* this = THIS;
+
+    if (this->dyna.actor.params == (s16)0xFFFF) {
+        if (!BgMoriHashigo_InitClasp(this, globalCtx2)) {
+            Actor_Kill(&this->dyna.actor);
+            return;
+        }
+    } else if (this->dyna.actor.params == 0) {
+        if (!BgMoriHashigo_InitLadder(this, globalCtx2)) {
+            Actor_Kill(&this->dyna.actor);
+            return;
+        }
+    }
+    this->objBankIndex = Object_GetIndex(&globalCtx2->objectCtx, OBJECT_MORI_TEX);
+    if (this->objBankIndex < 0) {
+        // Bank danger!
+        osSyncPrintf("Error : バンク危険！(arg_data 0x%04x)(%s %d)\n", this->dyna.actor.params,
+                     "../z_bg_mori_hashigo.c", 312);
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        BgMoriHashigo_SetupObjectCheck(this);
+        // (Forest Temple Ladder and its clasp)
+        osSyncPrintf("(森の神殿 梯子とその留め金)(arg_data 0x%04x)\n", this->dyna.actor.params);
+    }
+}
+
+void BgMoriHashigo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriHashigo* this = THIS;
+
+    if (this->dyna.actor.params == 0) {
+        DynaPolyInfo_Free(globalCtx2, &globalCtx2->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+    if (this->dyna.actor.params == (s16)0xFFFF) {
+        Collider_DestroyJntSph(globalCtx, &this->collider);
+    }
+}
+
+void BgMoriHashigo_SetupObjectCheck(BgMoriHashigo* this) {
+    this->actionFunc = BgMoriHashigo_ObjectCheck;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A2698.s")
+void BgMoriHashigo_ObjectCheck(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        if (this->dyna.actor.params == (s16)0xFFFF) {
+            BgMoriHashigo_SetupClasp(this);
+        } else if (this->dyna.actor.params == 0) {
+            BgMoriHashigo_SetupLadderWait(this);
+        }
+        this->dyna.actor.draw = BgMoriHashigo_Draw;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A2770.s")
+void BgMoriHashigo_SetupClasp(BgMoriHashigo* this) {
+    this->actionFunc = BgMoriHashigo_Clasp;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A27F8.s")
+void BgMoriHashigo_Clasp(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    if (this->hitTimer <= 0) {
+        if (this->collider.base.acFlags & 2) {
+            this->collider.base.acFlags &= ~2;
+            this->hitTimer = 10;
+        } else {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/BgMoriHashigo_Init.s")
+void BgMoriHashigo_SetupLadderWait(BgMoriHashigo* this) {
+    this->actionFunc = BgMoriHashigo_LadderWait;
```

Comment:
```suggestion
void BgMoriHashigo_SetupLadderBlocked(BgMoriHashigo* this) {
    this->actionFunc = BgMoriHashigo_LadderBlocked;
```
Maybe `Hooked` would be appropriate? or `Attached`
actually `Clasped` would make sense lol, and `Unclasped` instead of `Freed` for the other?

Reply:
Action names should remain infinitive, like `Wait` or `Fly`

Reply:
so, `Hang` here? 🤔 
I don't like seeing "Wait" everywhere tbh

Reply:
Wait seems fine for an actor that is just doing nothing until it receives a signal.

Reply:
yeah but a lot of actors do that 😅
would be nice to be a bit more specific imo but when it's too hard then nvm I guess